ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -511,10 +505,13 @@ private void addUpdateResources(String dcName, Map<String, Set<String>> partitio
           ArrayList<String> newInstances = new ArrayList<>(instanceSetInStatic);
           Collections.shuffle(newInstances);
           resourceIs.setPreferenceList(partitionName, newInstances);
+          // Existing resources may not have ANY_LIVEINSTANCE set as the numReplicas (which allows for different replication
+          // for different partitions under the same resource). So set it here.
+          resourceIs.setReplicas(ResourceConfig.ResourceConfigConstants.ANY_LIVEINSTANCE.toString());
           resourceModified = true;
         }
       }
-      if (resourceModified) {
+      if (!dryRun && resourceModified) {
```

Comment:
Correct me if I am wrong, if `!dryRun && resourceModified == true` and if `resourceIs.getPartitionSet().isEmpty() == false`, the extra partition has been removed from current resource, should we update the num of partition in current resource. Something like: `resourceIs.setNumPartitions( # of partition)`?

Reply:
It does not affect us really as Helix is fine with it and because we do not use a resource in any meaningful way. That said, I agree that this is a good idea, so have made the change.